### DISCUSSION TITLE
How to simply wrap a component like ActionIcon?

### DISCUSSION THREAD

**Discussion by Toshinaki** [mantinedev/mantine#2687] [Q&A] [answered]
I'm creating a page and an animated `ActionIcon` is repeatedly used, so I'm thinking of wrapping the `ActionIcon` for reuse.

I tried
```typescript
import { memo } from "react";
import { motion } from "framer-motion";
import { ActionIconProps, ActionIcon } from "@mantine/core";

const FadeActionIcon = (props: ActionIconProps): JSX.Element => (
  <ActionIcon
    component={motion.button}
    variants={{
      initial: { opacity: 0 },
      animate: { opacity: 1 },
    }}
    {...props}
  />
);

export default memo(FadeActionIcon) as typeof FadeActionIcon;
```

But then I could not do `<FadeActionIcon onClick={...}>...</FadeActionIcon >`, it complains "Property 'onClick' does not exist on type 'IntrinsicAttributes & ActionIconProps'."

How to do this properly?

**Comment by rtivital** [comment, marked as answer]
https://mantine.dev/guides/polymorphic/#create-your-own-polymorphic-components

**Reply by Toshinaki** [reply]
`import { PolymorphicComponentProps } from "@mantine/utils";`
then 
`PolymorphicComponentProps<"button", ActionIconProps>` did the job.

Though it'd be great to have a more elegant solution, like just a simple `ActionIconProps`
